### PR TITLE
Convert example to use Makie.jl instead of Plots.jl + add links to examples in Docs/Modules

### LINKED ISSUE #205: Change examples' workflow
I'm wondering whether we should follow the paradigm by Oceananigans.jl and in the examples, instead of animating the solutions online while time stepping, we demonstrate how we save snapshots in a `.jld2` file and then load it and produce the animation.

### LINKED ISSUE #209: Transition from Plots.jl to GLMakie.jl
I'm wondering whether we should transition all our example to use GLMakie.jl for plotting. In that way we can plot directly CuArrays without the need to transfer things back to the CPU.

Or we can have half of our examples use GLMakie.jl...

### LINKED ISSUE #261: `Plots` no longer accepts functions as `clims`
When I run `examples/multilayerqg_2layer.jl` on a fresh clone on Julia 1.6.2 I get an error:

```julia
julia> include("multilayerqg_2layer.jl")
[ Info: Precompiling GeophysicalFlows [44ee3b1c-bc02-53fa-8355-8e347616e15e]
WARNING: could not import Printf.ini_hex into BFloat16s
WARNING: could not import Printf.ini_HEX into BFloat16s
[ Info: FourierFlows will use 8 threads
[ Info: FourierFlows will use 8 threads
[ Info: Precompiling Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]
step: 0000, t: 0.0, cfl: 0.00, KE₁: 1.155e-08, KE₂: 5.483e-08, PE: 2.958e-09, walltime: 0.01 min
[ Info: Precompiling GR_jll [d2c73de3-f751-5644-a686-071e5b155ba9]
ERROR: LoadError: MethodError: no method matching iterate(::typeof(symlims))
Closest candidates are:
  iterate(::Union{LinRange, StepRangeLen}) at range.jl:664
  iterate(::Union{LinRange, StepRangeLen}, ::Int64) at range.jl:664
  iterate(::T) where T<:Union{Base.KeySet{var"#s79", var"#s78"} where {var"#s79", var"#s78"<:Dict}, Base.ValueIterator{var"#s77"} where var"#s77"<:Dict} at dict.jl:693
  ...
```

Is it the case that `Plots` no longer accepts function arguments to `clims`?

It's not so easy to fix the issue because some wizardry is used to update the plot data:

https://github.com/FourierFlows/GeophysicalFlows.jl/blob/dbd91a3a28ef9246c67e6f1f49e849640b8d3561/examples/multilayerqg_2layer.jl#L217-L220

discovered by @szy21

### LINKED ISSUE #269: Demonstrate save/load output functionality in an example
### Discussed in https://github.com/FourierFlows/GeophysicalFlows.jl/discussions/268

<div type='discussions-op-text'>

<sup>Originally posted by **parfenyev** November 19, 2021</sup>
Hello everyone,

I am studying an [example](https://github.com/FourierFlows/GeophysicalFlows.jl/blob/main/examples/twodnavierstokes_stochasticforcing.jl) from the documentation and I have a question: how can I save the result of calculation to a file and then load it from the file? I want to work with the time series of velocity field in the future (calculate the spectra, average them over time, etc.). An example of writing the results to a file and then reading them would help me a lot.

Thank you in advance! </div>

### PR DESCRIPTION
**Pull request by navidcy**
This PR converts all examples to use Makie for plots and animations.

Also, in the example `singlelayerqg_betaforced.jl` instead of animation on-the-fly, we demonstrate how we can save output in `.jld2` file and then load it and visualize it. This addresses #269.

Additionally, it fixes some of the links to the examples in the docs and in the README.

Resolves #209
Closes #261
Closes #269
Closes #205

### CODE REVIEW

File: examples/twodnavierstokes_stochasticforcing_budgets.jl [hunk 1]
```diff
@@ -141,114 +145,9 @@ Wᶻ = Diagnostic(TwoDNavierStokes.enstrophy_work,                       prob, n
...
-end
-nothing # hide
-
-
-
-
 # ## Time-stepping the `Problem` forward
 
-# Finally, we time-step the `Problem` forward in time.
+# We time-step the `Problem` forward in time.
```
Comment (jbisits) [inline examples/twodnavierstokes_stochasticforcing_budgets.jl:150]: We step the `Problem` forward in time.

File: examples/twodnavierstokes_stochasticforcing_budgets.jl [hunk 2]
```diff
@@ -266,13 +165,102 @@ end
...
+          title = "∇²ψ(x, y, μt=" * @sprintf("%.2f", μ * clock.t) * ")",
+          aspect = 1,
+          limits = ((-L/2, L/2), (-L/2, L/2)))
+
+heatmap!(ax, x, y, Array(vars.ζ);
+         colormap = :viridis, colorrange = (-25, 25))
+
+fig
+
+# And finaly, we plot the evolution of the energy and enstrophy diagnostics and all terms
```
Comment (jbisits) [inline examples/twodnavierstokes_stochasticforcing_budgets.jl:184]: And finaly -> Finally

**Review by jbisits — CHANGES_REQUESTED** [review]
Great stuff here @navidcy!! I added a few phrasing comments and clicked on Request changes but feel free to reject any updates to the phrasing if you think what is already in the text (or some other phrasing that comes to you..) is clearer.

File: docs/src/modules/surfaceqg.md [hunk 3]
```diff
@@ -96,6 +96,6 @@ Other diagnostic include: [`buoyancy_dissipation`](@ref GeophysicalFlows.Surface
 
 ## Examples
 
-- [`examples/surfaceqg_decaying.jl`](../../literated/surfaceqg_decaying/): A script that simulates decaying surface quasi-geostrophic flow with a prescribed initial buoyancy field, producing an animation of the evolution of the surface buoyancy.
+- [`examples/surfaceqg_decaying.jl`](@ref surfaceqg_decaying_example): Simulate decaying surface quasi-geostrophic flow with a prescribed initial buoyancy field, producing an animation of the evolution of the surface buoyancy.
```
[the remark this answers is not in the document]
Comment (navidcy) [reply docs/src/modules/surfaceqg.md:100]: Indeed.

File: examples/multilayerqg_2layer.jl [hunk 4]
```diff
@@ -12,13 +12,13 @@
 
 # ```julia
 # using Pkg
-# pkg"add GeophysicalFlows, Plots, Printf"
+# pkg"add GeophysicalFlows, CairoMakie, Printf"
 # ```
 
 # ## Let's begin
 # Let's load `GeophysicalFlows.jl` and some other needed packages.
```
[the remark this answers is not in the document]
Comment (navidcy) [reply examples/multilayerqg_2layer.jl:19]: Agreed.

File: examples/singlelayerqg_decaying_topography.jl [hunk 5]
```diff
@@ -158,63 +162,25 @@ if isfile(filename); rm(filename); end
...
-        framestyle = :box)
-
-  l = @layout Plots.grid(1, 2)
-  p = plot(pq, pψ, layout = l, size = (800, 360))
-  
-  return p
-end
+# We modify the figure with the initial state slightly. We add the topography contours and
+# also the time.
+
```
[the remark this answers is not in the document]
Comment (navidcy) [reply examples/singlelayerqg_decaying_topography.jl:174]: Thanks!

File: examples/twodnavierstokes_stochasticforcing_budgets.jl (hunk 1 above)
Comment (navidcy) [reply examples/twodnavierstokes_stochasticforcing_budgets.jl:150]: Done

File: examples/twodnavierstokes_stochasticforcing_budgets.jl (hunk 2 above)
Comment (navidcy) [reply examples/twodnavierstokes_stochasticforcing_budgets.jl]: Done

File: docs/src/modules/singlelayerqg.md [hunk 6]
```diff
@@ -110,10 +110,10 @@ Other diagnostic include: [`energy_dissipation`](@ref GeophysicalFlows.SingleLay
...
 
-- [`examples/singlelayerqg_betaforced.jl`](../../literated/singlelayerqg_betaforced/): A script that simulates forced-dissipative quasi-geostrophic flow on a beta plane demonstrating zonation. The forcing is temporally delta-correlated with isotropic spatial structure with power in a narrow annulus in wavenumber space with total wavenumber ``k_f``.
+- [`examples/singlelayerqg_betaforced.jl`](@ref singlelayerqg_betaforced_example): Simulate forced-dissipative quasi-geostrophic flow on a beta plane demonstrating zonation. The forcing is temporally delta-correlated with isotropic spatial structure with power in a narrow annulus in wavenumber space with total wavenumber ``k_f``.
 
-- [`examples/singlelayerqg_decay_topography.jl`](../../literated/singlelayerqg_decay_topography/): A script that simulates two dimensional turbulence (barotropic quasi-geostrophic flow with ``\beta=0``) above topography.
+- [`examples/singlelayerqg_decay_topography.jl`](@ref singlelayerqg_decay_topography_example): Simulate two dimensional turbulence (barotropic quasi-geostrophic flow with ``\beta=0``) above topography.
```
[the remark this answers is not in the document]
Comment (navidcy) [reply docs/src/modules/singlelayerqg.md:117]: What's the difference of $f$-plane from 2D turbulence? $f$ doesn't play any role in the streamfunction-vorticity inversion, at least not with doubly periodic boundary conditions. So even if in principle we set $f \ne 0$ then the solution won't differ from 2D turbulence.

**Comment by navidcy** [comment]
Thanks @jbisits -- did most of the changes. Can you have a look and approve?

**Comment by navidcy** [comment]
I upgraded FourierFlows.jl so we need to wait until it's merged for the CI to pass here...

**Comment by jbisits** [comment]
> Thanks @jbisits -- did most of the changes. Can you have a look and approve?

Yep can do!

**Comment by jbisits** [comment]
Should I wait for all the actions to finish before reviewing? Or just buildkite and CI?

File: README.md [hunk 7]
```diff
@@ -62,14 +62,14 @@ Some animations created with GeophysicalFlows.jl are [online @ youtube].
...
 ## Modules
 
-* `TwoDNavierStokes`: the two-dimensional vorticity equation.
-* `SingleLayerQG`: the barotropic or equivalent-barotropic quasi-geostrophic equation, which 
+* [`TwoDNavierStokes`](https://fourierflows.github.io/GeophysicalFlowsDocumentation/stable/modules/twodnavierstokes/): the two-dimensional vorticity equation.
+* [`SingleLayerQG`](https://fourierflows.github.io/GeophysicalFlowsDocumentation/stable/modules/singlelayerqg/): the barotropic or equivalent-barotropic quasi-geostrophic equation, which 
   generalizes `TwoDNavierStokes` to cases with topography, Coriolis parameters of the form 
   `f = f₀ + βy`, and finite Rossby radius of deformation.
-* `MultiLayerQG`: a multi-layer quasi-geostrophic model over topography and with the ability 
+* [`MultiLayerQG`](https://fourierflows.github.io/GeophysicalFlowsDocumentation/stable/modules/multilayerqg/): a multi-layer quasi-geostrophic model over topography and with the ability
```
Comment (jbisits) [inline README.md:69]: Don't need the and here.

**Review by jbisits — CHANGES_REQUESTED** [review]
One last thing

File: README.md (hunk 7 above)
Comment (navidcy) [reply README.md]: I rephrased.

**Review by jbisits — APPROVED** [review]

**Pull request merged**